File: 151-reverse-words-in-a-string/reverse-words-in-a-string.go
func reverseWords(s string) string {
s = strings.Join(strings.Fields(s), " ")
	strs := strings.Split(s, " ")

	e := len(strs) - 1
	for i := 0; i < e; i++ {
		strs[i], strs[e] = strs[e], strs[i]
		e--
	}
	s = strings.Join(strs, " ")
	return s
}

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution(object):
    def reverseWords(self, s):
        str1 = " ".join(s.split()).split()
        valid = [False] * len(str1)

        l = 0
        r = len(str1)-1

        while l < len(str1) and r > -1:
            if valid[l] == False and valid[r] == False:
                str1[l], str1[r] = str1[r], str1[l]
                valid[l] = True 
                valid[r] = True
            l += 1
            r -= 1
        
        return " ".join(str(item) for item in str1)
        """
        :type s: str
        :rtype: str
        """
        